crates/anvil/core/src/eth/transaction/mod.rs

```diff
@@ -141,6 +153,22 @@ impl EthTransactionRequest {
                     access_list: access_list.unwrap_or_default(),
                 }))
             }
+            // op-stack deposit
+            (Some(126), _, None, None, None) => {
```

Comment:
I think we're missing the where everything is None but deposit args are set, because atm it would be possible to end up with an eip1559 if deposit fields are present

https://github.com/foundry-rs/foundry/blob/master/crates/anvil/core/src/eth/transaction/mod.rs#L176-L176

all the other arms are also affected by this because we don't check for OP fields, so I wonder if we should add an additional check for the existence of OP fields first before entering the match

this entire logic is pretty horrible I admit...